src/test/ui/associated-types/associated-types-eq-3.stderr

```diff
@@ -19,7 +19,7 @@ error[E0271]: type mismatch resolving `<isize as Foo>::A == Bar`
 LL |     foo1(a);
    |     ^^^^ type mismatch resolving `<isize as Foo>::A == Bar`
    |
-note: expected this to be `Bar`
+note: expected struct `Bar`, found `usize`
```

Comment:
This seems like a regression, maybe there is some special-casing for `ValuePairs::Types` somewhere that we need to teach about `Term`, too? (ideally we'd remove `ValuePairs::Types` and `ValuePairs::Consts` entirely and use `Term` everywhere, but I fear that is a bigger PR ^^)

Reply:
I know I asked you to add `ValuePairs::Terms`, but maybe revert that and do the full change suggested above in a separate PR? That should give us the diagnostics back without burdening you more on this PR?

Reply:
Sounds good, will do